src/leveled_cdb.erl

```diff
@@ -386,63 +367,60 @@ cdb_deletepending(Pid, ManSQN, Inker) ->
 %% LastPosition could be the atom complete when the last key processed was at
 %% the end of the file.  last_key must be defined in LoopState.
 cdb_scan(Pid, FilterFun, InitAcc, StartPosition) ->
-    gen_fsm:sync_send_all_state_event(Pid,
-                                        {cdb_scan,
-                                            FilterFun,
-                                            InitAcc,
-                                            StartPosition},
-                                        infinity).
+    gen_statem:call(Pid,
+                    {cdb_scan, FilterFun, InitAcc, StartPosition},
+                    infinity).
 
 -spec cdb_lastkey(pid()) -> any().
 %% @doc
 %% Get the last key to be added to the file (which will have the highest
 %% sequence number)
 cdb_lastkey(Pid) ->
-    gen_fsm:sync_send_all_state_event(Pid, cdb_lastkey, infinity).
+    gen_statem:call(Pid, cdb_lastkey, infinity).
 
 -spec cdb_firstkey(pid()) -> any().
 cdb_firstkey(Pid) ->
-    gen_fsm:sync_send_all_state_event(Pid, cdb_firstkey, infinity).
+    gen_statem:call(Pid, cdb_firstkey, infinity).
 
 -spec cdb_filename(pid()) -> string().
 %% @doc
 %% Get the filename of the database
 cdb_filename(Pid) ->
-    gen_fsm:sync_send_all_state_event(Pid, cdb_filename, infinity).
+    gen_statem:call(Pid, cdb_filename, infinity).
 
 -spec cdb_keycheck(pid(), any()) -> probably|missing.
 %% @doc
 %% Check to see if the key is probably present, will return either
 %% probably or missing.  Does not do a definitive check
 cdb_keycheck(Pid, Key) ->
-    gen_fsm:sync_send_event(Pid, {key_check, Key}, infinity).
+    gen_statem:call(Pid, {key_check, Key}, infinity).
 
 -spec cdb_isrolling(pid()) -> boolean().
 %% @doc
 %% Check to see if a cdb file is still rolling
 cdb_isrolling(Pid) ->
-    gen_fsm:sync_send_all_state_event(Pid, cdb_isrolling, infinity).
+    gen_statem:call(Pid, cdb_isrolling, infinity).
 
 -spec cdb_clerkcomplete(pid()) -> ok.
 %% @doc
 %% When an Inker's clerk has finished with a CDB process, then it will call
 %% complete.  Currently this will prompt hibernation, as the CDB process may
-%% not be needed for a period.  
+%% not be needed for a period.
 cdb_clerkcomplete(Pid) ->
-    gen_fsm:send_all_state_event(Pid, clerk_complete).
+    gen_statem:cast(Pid, clerk_complete).
 
```

Comment:
The clerk_complete message can only be sent when in the `reader` state.  It is send only by the leveled_iclerk, and the leveled_iclerk only examines files in this state.  The state cannot be transitioned to delete_pending without the leveled_iclerk first having called clerk_complete - so there should be no race.  So handling this message should be done in the `reader` state only.

There is an issue otherwise with code coverage, in that there will no longer be 100% code coverage (and so spotting regression of code coverage will be hard).